x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/predicate/fulltext/MatchQueryPredicate.java

```diff
@@ -33,6 +36,25 @@ public MatchQueryPredicate(Source source, Expression field, String query, String
         this.field = field;
     }
 
+    public MatchQueryPredicate(Source source, Expression field, String query, Double boost, Fuzziness fuzziness) {
+        super(source, query, createOptions(boost, fuzziness), singletonList(field));
+        this.field = field;
+    }
+
+    private static String createOptions(Double boost, Fuzziness fuzziness) {
+        StringBuilder options = new StringBuilder();
+        if (boost != null) {
+            options.append(BOOST_OPTION).append("=").append(boost);
+        }
+        if (fuzziness != null) {
+            if (boost != null) {
+                options.append(";");
+            }
+            options.append(FUZZINESS_OPTION).append("=").append(fuzziness.asString());
```

Comment:
If we decide to go with `MatchQueryPredicate` I would actually remove this functionality of passing options as a string from `FullTextPredicate` and have options be sent in as a map. We could do this refactor as a separate PR which would be faster and easier to review.
The reason for this is that (1) we would avoid doing this type of string concatenation, only for it to be transformed back as a map and (2) when we add options for full text functions, we are certain we do not want them to be sent in as a string, it is not the user experience we want to offer.
On the other hand if we decide that instead of using the `MatchQueryPredicate` we should use the `match` function, I would actually remove  the `MatchQueryPredicate` at some point.

Reply:
Yes, this is ugly. I wanted to check on the overall approach before refactoring this though 👍 . Both of the options you mention LGTM.

Reply:
The query is built in `EsqlExpressionTranslators`, consider building the options in `EsqlExpressionTranslators` together with the query.